src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs

```diff
@@ -246,6 +249,17 @@ public bool EnableTSplineIsChecked
                 RaisePropertyChanged(nameof(EnableTSplineIsChecked));
             }
         }
+
+        /// <summary>
+        /// Gets the different Python Engine versions availables from PythonNodeModels.dll
+        /// </summary>
+        /// <returns>Strings array with the different names</returns>
+        private string[] GetPythonEngineOptions()
+        {
+            var assembly = Assembly.LoadFile(AppDomain.CurrentDomain.BaseDirectory + "nodes\\PythonNodeModels.dll");
```

Comment:
I think this is likely to cause fewer conflicts / be better isolated
https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assembly.reflectiononlyloadfrom?view=net-5.0

if you give this a read:
https://docs.microsoft.com/en-us/dotnet/framework/deployment/best-practices-for-assembly-loading#:~:text=of%20load%20contexts-,Avoid%20binding%20on%20partial%20assembly%20names,to%20the%20default%20load%20context

note the big warnings on top:
if you use LoadFile you will potentially be loading that assembly into multiple contexts.

My other suggestion, which I think I was not clear enough about -  was to just look for the type in the already loaded assemblies - instead of trying to load it manually. If it's not already loaded, give up to avoid all the issues with potentially loading an assembly twice into different contexts, that causes all sorts of weird runtime behavior.

Reply:
Ah I see, I will give those links a look and make some changes, thanks for the explanation

Reply:
@Astul-Betizagasti some more thoughts :)

1. I think the safest thing is still to just check if the type is already around in the loaded assemblies - it should be since this is UI code and the engine should have been loaded when the DynamoModel started up.
2. I don't know if the reflectionOnly context counts as a real context - in which case maybe it's just as bad as the non default context? I'm not sure if we normally load the python assembly into the default or loadFrom contexts.
3. To be super safe if you want to load the assembly for certain - is to load the assembly into a new app domain, check the enum, then dispose the app domain and unload the assembly, this will make sure it is not interfering with dynamo. This could be a good pattern for dynamic operations we may need to do in the future. For example - inspecting a library of nodes before loading them.
4. I also think hardcoding the path is a bad idea - it's fragile and will break when we modify the repo structure, but we won't get compile errors - maybe you can use the pathManager to get this path - I think the `nodes` folder is a known location in the `PathManager`

Reply:
Updated

Reply:
@mjkkirschner What do you think of the update?

Reply:
Hi @QilongTang @Astul-Betizagasti - I think the PR is good as far as porting over the settings functionality to the preferences dialog.

I do think that the dynamic nature of the python engines needs to be addressed sooner rather than later, and at that point, this preference dialog should be changed to take advantage of it but that seems out of scope for this PR.

some things to think about for that work:
* when we no longer ship IronPython 2.x - how will the preference and enum change?
* If a user wrote their own ironpython3.x engine (it seems to be getting further along) - https://github.com/IronLanguages/ironpython3 - how would the preferences and engine dropdown react? - How would they add their implementation?
* What happens if a users preferences set a default python engine that is not loaded?